madanalysis/layout/histogram.py

```diff
@@ -1,125 +1,144 @@
+        self.summary.sumw = np.clip(self.positive.sumw - self.negative.sumw, 0, None)
 
         # sumw2
-        self.summary.sumw2     = self.positive.sumw2     - self.negative.sumw2
-        if self.summary.sumw2<0:
-            self.summary.sumw2=0
+        self.summary.sumw2 = np.clip(self.positive.sumw2 - self.negative.sumw2, 0, None)
 
         # sumwx
-        self.summary.sumwx     = self.positive.sumwx     - self.negative.sumwx
+        self.summary.sumwx = self.positive.sumwx - self.negative.sumwx
         # no correction on it
 
         # sumw2x
-        self.summary.sumw2x    = self.positive.sumw2x    - self.negative.sumw2x
+        self.summary.sumw2x = self.positive.sumw2x - self.negative.sumw2x
         # no correction on it
 
         # underflow
-        self.summary.underflow = self.positive.underflow - self.negative.underflow
-        if self.summary.underflow<0:
-            self.summary.underflow=0
+        self.summary.underflow = np.clip(
+            self.positive.underflow - self.negative.underflow, 0, None
+        )
 
         # overflow
-        self.summary.overflow  = self.positive.overflow  - self.negative.overflow
-        if self.summary.overflow<0:
-            self.summary.overflow=0
+        self.summary.overflow = np.clip(
+            self.positive.overflow - self.negative.overflow, 0, None
+        )
+
+        # compute mean and uncertainties for the statistics
+        # ! @jackaraz: this portion of the code should be changed to accomodate different types of
+        # ! PDF + scale unc combination for now its just mean and std
+        for tp in [
+            "nevents",
+            "nentries",
+            "sumw",
+            "sumw2",
+            "sumwx",
+            "sumw2x",
+            "underflow",
+            "overflow",
+        ]:
+            # compute unc shape: (lower, upper)
+            std = float(np.std(getattr(self.summary, tp)))
+            setattr(self.summary, tp + "_unc", (std, std))
+            # compute mean
+            setattr(self.summary, tp, float(np.mean(getattr(self.summary, tp))))
 
         # Data
         data = []
-        for i in range(0,len(self.positive.array)):
-            data.append(self.positive.array[i]-self.negative.array[i])
-            if data[-1]<0:
-                self.warnings.append(\
-                    'dataset='+dataset.name+\
-                    ' -> bin '+str(i)+\
-                    ' has a negative content : '+\
-                    str(data[-1])+'. This value is set to zero')
-                data[-1]=0
-        self.summary.array = data[:] # [:] -> clone of data
+        for i, array in enumerate(self.positive.array):
+            data.append(np.array(array) - np.array(self.negative.array[i]))
+            if np.any(data[-1] < 0):
+                self.warnings.append(
+                    f"dataset={dataset.name} -> bin {i} has a negative content : "
+                    f"{str(data[-1])}. This value is set to zero."
```

Comment:
Maybe time to change this behaviour in something better.

Reply:
such as?

Reply:
Maybe plot the abs and using two different line styles for negative and positive contributions? As done in the MG5 plotter.

Reply:
I rarely use other people's plotters; I don't know what mg5 plotter does. Why are we separating positive and negative contributions? A generic plotter will already be very messy as is; adding more lines will make it even harder, IMHO.

Reply:
I was more talking about the style of the plotter: slid lines for positive bins and dashed lines for negative bins. Bins are then shown as absolute values.